File: src/my_cpp_pkg/src/sum_server.cpp
#include "example_interfaces/srv/add_two_ints.hpp"
#include "rclcpp/rclcpp.hpp"
#include <memory>
#include <rclcpp/executors.hpp>
#include <rclcpp/utilities.hpp>

class SumServer : public rclcpp::Node {
public:
  SumServer() : rclcpp::Node("sum_server_cpp") {
    this->server_ = this->create_service<example_interfaces::srv::AddTwoInts>(
        "sum_up", std::bind(&SumServer::callbackService, this,
                            std::placeholders::_1, std::placeholders::_2));
    RCLCPP_INFO(this->get_logger(), "Server Started");
  }

private:
  void callbackService(
      const example_interfaces::srv::AddTwoInts::Request::SharedPtr request,
      example_interfaces::srv::AddTwoInts::Response::SharedPtr response) {

    response->sum = request->a + request->b;
    RCLCPP_INFO(this->get_logger(), "SUMMATION PERFORMED %d",
                (int)response->sum);
  }
  //   rclcpp::Client<example_interfaces::srv::AddTwoInts>::SharedPtr client_;
  rclcpp::Service<example_interfaces::srv::AddTwoInts>::SharedPtr server_;
};

int main(int argc, char **argv) {
  rclcpp::init(argc, argv);
  auto node = std::make_shared<SumServer>();
  rclcpp::spin(node);
  rclcpp::shutdown();
}

File: src/my_py_pkg/my_py_pkg/sum_server.py
# import rclpy
# from rclpy.node import Node
# from example_interfaces.srv import AddTwoInts


# class SumServer(Node):
#     def __init__(self):
#         super().__init__("sum_server")
#         self.server_ = self.create_service(AddTwoInts, "sum_up", self.callback_sum_up)
#         self.get_logger().info("Sum server started")

#     def callback_sum_up(
#         self, request: AddTwoInts.Request, response: AddTwoInts.Response
#     ):
#         sum = request.a + request.b
#         response.sum = sum
#         self.get_logger().info("SUMMATION DONE " + str(response.sum))
#         return response


# def main(args=None):
#     rclpy.init(args=args)
#     node = SumServer()
#     rclpy.spin(node)
#     rclpy.shutdown()


# if __name__ == "__main__":
#     main()

import rclpy
from rclpy.node import Node
from example_interfaces.srv import AddTwoInts


class SumServer(Node):
    def __init__(self):
        super().__init__("sum_server")
        self.service_ = self.create_service(AddTwoInts, "sum_up", self.serviceCallback)
        self.get_logger().info("Service Started")

    def serviceCallback(
        self, request: AddTwoInts.Request, response: AddTwoInts.Response
    ):
        sum = request.a + request.b
        response.sum = sum
        self.get_logger().info("Sum = " + str(sum))
        return response


def main(args=None):
    rclpy.init(args=args)
    node = SumServer()
    rclpy.spin(node)
    rclpy.shutdown


if __name__ == "__main__":
    main()
